sql/src/main/java/org/apache/druid/sql/calcite/parser/DruidSqlInsert.java

```diff
@@ -62,10 +68,14 @@ public DruidSqlInsert(
         insertNode.getSource(),
         insertNode.getTargetColumnList()
     );
-    Preconditions.checkNotNull(partitionedBy); // Shouldn't hit due to how the parser is written
+    if (partitionedBy == null) {
+      throw new ParseException("INSERT statements should specify PARTITIONED BY clause explictly");
```

Comment:
The word "must" is better than "should" here. (RFC 2119 doesn't govern here, but is still a helpful guide for technical writing. "Should" means "recommended, but not strictly required".)

Reply:
Updated the error message. Thanks for the tip.